### ISSUE TITLE
Support keychain agents

### ISSUE THREAD

**Issue by nonspecialist** [realestate-com-au/credulous#36]
OSX and most linux distros now have some kind of keychain agent. We should at least consider supporting them.

**Comment by lox** [comment]
Yeah, FWIW, we built our own that relies on this. https://github.com/99designs/aws-vault
